pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/BrokerOperabilityMetrics.java

```diff
@@ -60,8 +60,8 @@ public List<Metrics> getMetrics() {
         return metricsList;
     }
 
-    private void generate() {
-        metricsList.add(getOldTopicLoadMetrics());
+    private synchronized void generate() {
```

Comment:
If we always generate a brand new `metricsList`, maybe we can move it to a local variable and merge `generate` into `getMetrics`? This can also drop the `synchronized` modifier.

Reply:
fixed